KEP-1959 Update document for promoting loadbalancerclass to beta (#28051)
kubernetes/website: content/en/docs/concepts/services-networking/service.md

#### Specifying class of load balancer implementation {#load-balancer-class}

{{< feature-state for_k8s_version="v1.21" state="beta" >}}

> Starting in v1.21, you can optionally specify the class of a load balancer implementation for

Reviewer: ```suggestion
From Kubernetes v1.21 onwards, you can optionally specify the class of a load balancer implementation for
```

Author: Thanks! Updated 😄

#### Specifying class of load balancer implementation {#load-balancer-class}

> {{< feature-state for_k8s_version="v1.21" state="beta" >}}

Reviewer 2: cc @sftim shouldn't this be v1.22 ?

Reviewer: Yes, it should.

> From Kubernetes v1.21 onwards, you can optionally specify the class of a load balancer implementation for

Reviewer 3: I think this should say v1.22 because that's when it's on by default

Reviewer: We should say that it's available from v1.21 and also say that the feature gate is enabled by default from v1.22 onwards; that'll match how the rest of the docs tackle this sort of thing.

`LoadBalancer` type of Service by setting the field `spec.loadBalancerClass`.
By default, `spec.loadBalancerClass` is `nil` and a `LoadBalancer` type of Service uses

> the cloud provider's default load balancer implementation. 

Reviewer: You _can_ do this. Do we recommend that you should? Or is it truly optional?

Author: This is to provide a capability for users to disable default load balancer. 
So, this is truly optional. 
And it is for the users who want to use other load balancer implementations, they can set this value to disable default cloud providers' load balancer implementation.

Reviewer 4: @sftim do you have any comments on this, or can we move forward and merge this?

Reviewer: Let's explain that this change enables you to use a load balancer technology other than the cloud provider default. I don't think that's obvious.

> the cloud provider's default load balancer implementation. 

Reviewer 3: We probably want to specify this is only true when cloud provider integration is enabled:
```
By default, `spec.loadBalancerClass` is `nil` and a `LoadBalancer` type of Service uses
the cloud provider's default load balancer implementation if the cluster is configured with
a cloud provider using the `--cloud-provider` component flag.
```

or similar

The value of `spec.loadBalancerClass` must be a label-style identifier,
with an optional prefix such as "`internal-vip`" or "`example.com/internal-vip`".
Unprefixed names are reserved for end-users.

> You must enable the `ServiceLoadBalancerClass` feature gate to use this field.

Reviewer: We should also update https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ - has that happened in a different pull request?

Author: Yes, that already merged in this PR: #28644
